### ISSUE TITLE
What is a security model for self-hosted runners?

### ISSUE THREAD

**Issue by mdvorak** [cs-poc/actions-review#1]
a.    Is action code executed in sandbox, chroot, container, or at least random uid?
  b.    Does hold the same for raw shell commands?

_Original answer_
> The action code for a self hosted runner is executed in the same context you are running the runner as. We don't run in a VM for the self hosted runner. Since its a simple executable, then you can isolate its execution pretty easily. For the Docker workflow, it is executed in an isolated container which is spined from the VM.

**Comment by mdvorak** [comment]
Ok, so there is no isolation at all. 

1. Azure-based runners solve this by being one-use only. I can imagine similar pattern, is there a way to actually kill runner after each job execution? E.g. k8s would then restart the pod with clean data. Some hook called from the runner would be great for this. 
1. Another nice option would be if all jobs would actually execute in a container instead on the host. Performance overhead is negligible. I understand this is not easy to implement, but security would be much better.
1. Is it possible to use containers, but without docker? docker.sock is hard to secure, podman would be much better (and should work the same). On disposable runners with sudo this is not an issue, but on persistent runner it is a great problem.

To sum it up, self-hosted runners are now either very limited in what they can do, or very unsecure. And without our runners, we cannot leverage GHA as CI tool at all. Most of our services are in internal network.
